### PR TITLE
Add TypeScript example

### PR DESCRIPTION
**Pull request by javonharper**
Adds a `typescript.ts` example file for getting started with TypeScript.

In order to be used with TypeScript, we need to add type support for methods in our library. 
Submitted a PR to DefinitelyTyped here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31976.

Once this is  merged, we can remove the iopipe.d.ts file in this changeset and add instructions for adding the types via a `yarn add '@types/iopipe` call.

### CODE REVIEW

File: node/package.json [hunk 1]
```diff
@@ -20,7 +20,11 @@
   "homepage": "https://github.com/iopipe/examples#readme",
   "dependencies": {
     "@iopipe/iopipe": "^1.5.0",
+    "@types/aws-lambda": "^8.10.17",
+    "@types/iopipe__iopipe": "^1.12.0",
```
Comment (kolanos) [inline node/package.json]: We also control the `iopipe` package. Are we able to do something like `@types/iopipe` as well?

File: node/typescript.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import lambda = require('aws-lambda');
```
Comment (kolanos) [inline node/typescript.ts]: Why use `require()` here?

File: node/package.json (hunk 1 above)
Comment (javonharper) [reply node/package.json]: Yup we can add something to types/iopipe entry if that makes sense. But that only really makes sense if we had a library that could be installed at `iopipe` (non namespaced). I think `iopipe` is deprecated and points to `@iopipe/iopipe`.

File: node/typescript.ts (hunk 2 above)
Comment (javonharper) [reply node/typescript.ts]: Typescript imports and exports work slightly differently as up until a latest version you could only export one big namespace with a set of members or one default export, but not both. 

From the docs: `


From the TS docs: "When exporting a module using export =, TypeScript-specific import module = require("module") must be used to import the module."

File: node/package.json (hunk 1 above)
Comment (kolanos) [reply node/package.json]: Yes, i is "deprecated", but we're still publishing releases to it. So if that's what it takes to get `@types/iopipe` then maybe we just continue to support it?

**Pull request merged**